docs/hugo/content/design/ADR-2024-02-Upstream-Deletion/_index.md

```diff
@@ -0,0 +1,52 @@
+---
+title: 2024-02 Upstream Deletion
+---
+
+What do we do if/when a resource supported by ASO v2 is deleted upstream - removed from the Azure OpenAPI repository? 
+
+
+## Context
+
+In July 2022, we encountered a problem when the preview versions of the `Microsoft.Auithorization` service were deleted from the [azure-resource-manager-schemas](https://github.com/Azure/azure-resource-manager-schemas) repository that ASO consumed at the time. This orphaned the `RoleAssignment` resource in ASO, which would have been a breaking change for our users.
+
+Since that time, ASO has migrated to depend soley on the [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs) repo (see [PR#2323](https://github.com/Azure/azure-service-operator/pull/2323), merged December 2022) so the problem with `RoleAssignment` went away.
+
+However, we decided that this question still needed to be considered, as it's likely to come up again in the future and we need to have a plan in place for how to handle it.
```

Comment:
I don't know that it's actually that likely.

```suggestion
However, we decided that this question still needed to be considered, as it may come up again in the future and we need to have a plan in place for how to handle it.
```

Reply:
Oh, you even allude to this below (it's not likely), but we wanna document what to do anyway.

Reply:
Done.